snabbdom/src/main/scala/snabbdom/init.scala

```diff
@@ -404,9 +404,11 @@ object init {
       if (oldVnode != vnode) {
 
         cbs.update.foreach(hook => hook(oldVnode, vnode))
-        vnode.data.hook
-          .flatMap(_.update)
-          .foreach(hook => hook(oldVnode, vnode))
+
+        if (vnode.text.isEmpty || vnode.text != oldVnode.text)
+          vnode.data.hook
+            .flatMap(_.update)
+            .foreach(hook => hook(oldVnode, vnode))
```

Comment:
AFAICT in our case this becomes an optimization actually, since we were running these update hooks for text nodes already.